### ISSUE TITLE
Wrong time on event params for `eventClick` and other callbacks

### ISSUE THREAD

**Issue by paulyoung5** [fullcalendar/fullcalendar#5354]
When using `timeZone: local`, events received as parameters to FullCalendar callbacks have the wrong time, off by a value of the timezone offset.

It seems as though FC applies the timezone offset to the Date object internally, before it's received by callbacks.

I first discovered the issue while using the `fullcalendar-vue` component, but have since reproduced this with the vanilla JS package.

For the examples below and incase relevant, I'm in BST.

----

#### **Example 1:** `eventClick`

User clicks on a 5pm to 6pm event, and the event's `start` and `end` properties are logged to the console.

##### Expected behaviour
`start` should be 5pm BST, `end` should be 6pm BST

##### Actual behaviour
`start` is 6pm BST, `end` is 7pm BST

#### **Example 2:** `eventDrop`

User drags event to a new start time of 3pm, and the event's `start` and `end` properties are logged to the console.

##### Expected behaviour
`start` is 3pm BST

##### Actual behaviour
`start` is 4pm BST

---

**Reduced test case:**
https://codesandbox.io/s/summer-lake-zzwqc?file=/src/index.js

---

## GIF

![Screen Recording 2020-04-13 at 05 45 pm](https://user-images.githubusercontent.com/6493590/79139669-a0932200-7dae-11ea-9ab5-bf4bf8735ed2.gif)

**Comment by paulyoung5** [comment]
This seems linked to this line of code:
https://github.com/fullcalendar/fullcalendar/blob/master/packages/core/src/datelib/marker.ts#L209

When I start dragging an 8pm event, it receives an array with a time of 8pm BST - but returns a date object for 9pm BST.

**Comment by acerix** [comment]
The start time is a property of the event so you can access it like `event.start` which has the correct timezone. 

https://codesandbox.io/s/thirsty-sunset-wwqxo

**Comment by LeoZandvliet** [comment]
@acerix 
I actually experience the same problem with v5.6. Events show correct in the calendar, but the `event.start` in the `eventClick` is shifted by the timezone offset.
Are you willing to look into this? Do I need to open a new issue?

I've made a reproducable example with 3 events each with different formatted start string.
Once the document is ready, the console logs a simple expectation., On `eventClick` an alert is shown with actual and expected result.
Am I wrong?

[https://codepen.io/lzandvliet/pen/eYwQwpd](https://codepen.io/lzandvliet/pen/eYwQwpd)
